File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    private double seconds;
    private static final double EARTH_PERIOD = 365.25;

    private enum Planet {
        MERCURY(0.2408467),
        VENUS(0.61519726),
        EARTH(1.),
        MARS(1.8808158),
        JUPITER(11.862615),
        SATURN(29.447498),
        URANUS(84.016846),
        NEPTUNE(164.79132);

        private double earthYear;

        Planet(double earthYear) {
            this.earthYear = earthYear;
        }
    }

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double getSeconds() {
        return this.seconds;
    }

    double secondByYear(double earthYear) {
        return earthYear * EARTH_PERIOD * 24 * 60 * 60;
    }

    double onEarth() {
        return this.seconds / secondByYear(Planet.EARTH.earthYear);
    }

    double onMercury() {
        return this.seconds / secondByYear(Planet.MERCURY.earthYear);
    }

    double onVenus() {
        return this.seconds / secondByYear(Planet.VENUS.earthYear);
    }

    double onMars() {
        return this.seconds / secondByYear(Planet.MARS.earthYear);
    }

    double onJupiter() {
        return this.seconds / secondByYear(Planet.JUPITER.earthYear);
    }

    double onSaturn() {
        return this.seconds / secondByYear(Planet.SATURN.earthYear);
    }

    double onUranus() {
        return this.seconds / secondByYear(Planet.URANUS.earthYear);
    }

    double onNeptune() {
        return this.seconds / secondByYear(Planet.NEPTUNE.earthYear);
    }

}

File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private double _seconds;
    private static double EARTH_YEAR_IN_SECOND = 31557600;

    private struct Planet
    {
        public const double Earth = 1;
        public const double Mercury = 0.2408467;
        public const double Venus = 0.61519726;
        public const double Mars = 1.8808158;
        public const double Jupiter = 11.862615;
        public const double Saturn = 29.447498;
        public const double Uranus = 84.016846;
        public const double Neptune = 164.79132;
    }

    public SpaceAge(int seconds)
    {
        _seconds = seconds;
    }

    public double OnEarth()
    {
        return GetOrbitalAge(Planet.Earth);
    }

    public double OnMercury()
    {
        return GetOrbitalAge(Planet.Mercury);
    }

    public double OnVenus()
    {
        return GetOrbitalAge(Planet.Venus);
    }

    public double OnMars()
    {
        return GetOrbitalAge(Planet.Mars);
    }

    public double OnJupiter()
    {
        return GetOrbitalAge(Planet.Jupiter);
    }

    public double OnSaturn()
    {
        return GetOrbitalAge(Planet.Saturn);
    }

    public double OnUranus()
    {
        return GetOrbitalAge(Planet.Uranus);
    }

    public double OnNeptune()
    {
        return GetOrbitalAge(Planet.Neptune);
    }

    public double GetOrbitalAge(double planetOrbitalYear)
    {
        return (_seconds / EARTH_YEAR_IN_SECOND) / planetOrbitalYear;
    }
}